public/background.js

```diff
@@ -1,69 +1,55 @@
 /*global chrome*/
-// let bookmarkTree;
-let flattenedBookmarks = [], bookmarkCreationDates = [], bookmarkUrls = [], bookmarkFolderTree = [];
 
-const flattenNode = (node, result, bookmarkCreationDates, bookmarkUrls, bookmarkFolderTree, parentFolder) => {
-    if (node.children) {
-      //if it is a folder
-      let folderObj={};  
-      folderObj.id=node.id;
-      folderObj.children=[];
-      folderObj.dateAdded= node.dateAdded;
-      folderObj.index= node.index;
-      folderObj.parentId=node.parentId;
-      folderObj.title=node.title;
-      folderObj.isSelected=false;
-      folderObj.isOpen=false;
-      folderObj.parent = parentFolder;
-      bookmarkFolderTree.push(folderObj);
-  
-      node.children.forEach(child => {
-        //if it is a bookmark
-        if (child.url && child.title) {
-          result.push(
-              {
-                title: child.title,
-                url:child.url,
-                dateAdded:child.dateAdded,
-                id:child.id,
-                index:child.index,
-                parentId:child.parentId,
-                category:node.title
-              }
-          );
-          //data for bookmark analytics TODO need to think of optimization
-          bookmarkCreationDates.push(child.dateAdded);
-          bookmarkUrls.push(child.url);
-        }
-        flattenNode(child, result, bookmarkCreationDates, bookmarkUrls, folderObj.children, folderObj);
-      });
-      
+const flattenNode = (node, result = [], bookmarkCreationDates = [], bookmarkUrls = [], bookmarkFolderTree = [], parentFolder) => {
```

Comment:
inilializing flattenedBookmarks = [], bookmarkCreationDates = [], bookmarkUrls = [], bookmarkFolderTree = [] here will cause the variables to initialize, everytime the recrusion is called.
Note, the recursion is called at line 33.

Reply:
Default parameter in method means that if `flattenMode` has one parameter, will be `(node, [])`, but if you set second parameter will be `(node, result)`